src/storage/btree_load.c

```diff
@@ -3878,3 +3750,457 @@ btree_get_perf_btree_page_type (THREAD_ENTRY * thread_p, PAGE_PTR page_ptr)
     }
   return PERF_PAGE_BTREE_ROOT;
 }
+
+/*
+ * btree_load_check_fk () -
+ *
+ *    return:				- NO_ERROR or error code.
+ *    load_args_local(in):		- Information about the foreign key
+ *    sort_args_local(in):		- Information about the primary key
+ *
+ */
+int
+btree_load_check_fk (THREAD_ENTRY * thread_p, const LOAD_ARGS * load_args, const SORT_ARGS * sort_args)
+{
+  DB_VALUE fk_key, pk_key;
```

Comment:
Rename for clarity: for fk/pk_key_cnt and fk_pk_header add node_ inside name (e.g. fk_node_key_cnt, fk_node_header). They may be confused for index total count and header.

Remove unused advance_direction.